Snippets/scale-font.py

```diff
@@ -0,0 +1,273 @@
+
+def scaleLookupType5(subTable, UPM, target_UPM):    
+    for markRecord in subTable.MarkArray.MarkRecord:     
+        markRecord.MarkAnchor.XCoordinate = scaleValue_toUPM(markRecord.MarkAnchor.XCoordinate, UPM, target_UPM)
+        markRecord.MarkAnchor.YCoordinate = scaleValue_toUPM(markRecord.MarkAnchor.YCoordinate, UPM, target_UPM)
+    for ligatureAttach in subTable.LigatureArray.LigatureAttach:
+        for componentRecord in ligatureAttach.ComponentRecord:
+            for ligatureAnchor in componentRecord.LigatureAnchor:
+                ligatureAnchor.XCoordinate = scaleValue_toUPM(ligatureAnchor.XCoordinate, UPM, target_UPM)
+                ligatureAnchor.YCoordinate = scaleValue_toUPM(ligatureAnchor.YCoordinate, UPM, target_UPM)
+                
+def scaleLookupType6(subTable, UPM, target_UPM): 
+    for mark1Record in subTable.Mark1Array.MarkRecord:
+        mark1Record.MarkAnchor.XCoordinate = scaleValue_toUPM(mark1Record.MarkAnchor.XCoordinate, UPM, target_UPM)
+        mark1Record.MarkAnchor.YCoordinate = scaleValue_toUPM(mark1Record.MarkAnchor.YCoordinate, UPM, target_UPM)
+    for mark2Record in subTable.Mark2Array.Mark2Record:
+        for mark2Anchor in mark2Record.Mark2Anchor:
+            mark2Anchor.XCoordinate = scaleValue_toUPM(mark2Anchor.XCoordinate, UPM, target_UPM)
+            mark2Anchor.YCoordinate = scaleValue_toUPM(mark2Anchor.YCoordinate, UPM, target_UPM)
+
+def scaleLookupType7(subTable, UPM, target_UPM):
+    pass 
+    
+def scaleLookupType8(subTable, UPM, target_UPM):
+    pass        
+
+def scaleFont(tt, target_UPM):
+    UPM = tt['head'].unitsPerEm
+    glyphOrder = tt.getGlyphOrder()
+
+    # Data related to UPM
+    #################################
+    # HEAD
+    tt['head'].unitsPerEm = scaleValue_toUPM(tt['head'].unitsPerEm, UPM, target_UPM)
+    tt['head'].xMin = scaleValue_toUPM(tt['head'].xMin, UPM, target_UPM)
+    tt['head'].yMin = scaleValue_toUPM(tt['head'].yMin, UPM, target_UPM)
+    tt['head'].xMax = scaleValue_toUPM(tt['head'].xMax, UPM, target_UPM)
+    tt['head'].yMax = scaleValue_toUPM(tt['head'].yMax, UPM, target_UPM)
+    #################################
+    # HHEA
+    tt['hhea'].ascent = scaleValue_toUPM(tt['hhea'].ascent, UPM, target_UPM)
+    tt['hhea'].descent = scaleValue_toUPM(tt['hhea'].descent, UPM, target_UPM)
+    tt['hhea'].lineGap = scaleValue_toUPM(tt['hhea'].lineGap, UPM, target_UPM)
+    tt['hhea'].advanceWidthMax = scaleValue_toUPM(tt['hhea'].advanceWidthMax, UPM, target_UPM)
+    tt['hhea'].minLeftSideBearing = scaleValue_toUPM(tt['hhea'].minLeftSideBearing, UPM, target_UPM)
+    tt['hhea'].minRightSideBearing = scaleValue_toUPM(tt['hhea'].minRightSideBearing, UPM, target_UPM)
+    tt['hhea'].xMaxExtent = scaleValue_toUPM(tt['hhea'].xMaxExtent, UPM, target_UPM)
+    #################################
+    # print tt['hhea'].numberOfHMetrics #Number of entries of the hmtx table
+    # print len(glyphOrder)
```

Comment:
please get rid of stray commented-out lines like these

Reply:
right